### PR TITLE
ADD customer is now functional.  Customer has a foreign key to User.

### PR DESCRIPTION
**Pull request by edcomstock**
customer_listing.html will be implemented to view a list of customers,
but for now it is a placeholder.

### CODE REVIEW

File: msc/static/js/AddressCopy.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+function FillBilling(f) {
```
Comment (dnascimb) [inline msc/static/js/AddressCopy.js]: We didn't discuss this before but probably a good practice to make any of the JS files that correspond to a specific view, the same name as the view.

File: msc/views.py [hunk 2]
```diff
@@ -341,4 +342,113 @@ def validUserRequest(request):
...
+    if not session.get('logged_in'):
+        abort(401)
+
+    error = None
+    if not validCustomerAdd(request):
+        error = 'Invalid data entered'
+        print('invalid data entered')
+        return render_template('new_customer.html', error=error)
+
+    print('start add process...')
```
Comment (dnascimb) [inline msc/views.py]: best to comment or remove print statements before commiting

File: msc/static/js/AddressCopy.js (hunk 1 above)
Comment (edcomstock) [inline msc/static/js/AddressCopy.js]: Good point.  I thought about that but considered this to be generically
reusable by any page that had addresses to clone (mailing, shipping,
physical)

File: msc/views.py (hunk 2 above)
Comment (edcomstock) [inline msc/views.py]: Good rule, I'll do that going forward.

**Comment by edcomstock** [comment]
On to the next evolution...

**Pull request merged**